pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -82,4 +84,38 @@ var _ = Describe("Qemu agent poller", func() {
 			Expect(agentStore.AgentUpdated).ToNot(Receive())
 		})
 	})
+
+	Context("PollerWorker", func() {
+		const fakeAgentCommandName = "foo"
+
+		It("executes the agent commands at least once", func() {
+			w := PollerWorker{
+				CallTick:      1,
+				AgentCommands: []AgentCommand{fakeAgentCommandName},
+			}
+			c := make(chan struct{})
+
+			var execCounter int
+			go w.Poll(func(commands []AgentCommand) { execCounter++ }, c)
+			close(c)
```

Comment:
this test might be flaky
https://github.com/kubevirt/kubevirt/pull/4512/commits/a2fbf7e0b67a259040ea54bec4a978e5bb1e1737#diff-510b183b2d5d079863708a79f37a82f552d485d8c3e6e10e5e2eefecfc946edaR202

select takes the first event (if both at the same time it choose one random),
if the close happen before the Poll go routine got the timer, it wont run even one time,
and the test would flake, i would let it some time before closing as a context switch can happen here between the go routines.

2nd test might as well suffer, but didnt look on it deep, first lets discuss on this one please.